pkg/daemon/controller/controller.go

```diff
@@ -377,6 +359,44 @@ func (sc *ServiceController) resolvePowershellVariables(svc servicescm.Service)
 	return make(map[string]string), nil
 }
 
+// NewDirectClient creates and returns an authenticated client that reads directly from the API server.
+// It also returns the config and scheme used to created the client.
+func NewDirectClient(apiServerURL, saCA, saToken string) (client.Client, *rest.Config, *runtime.Scheme, error) {
+	cfg, err := config.FromServiceAccount(apiServerURL, saCA, saToken)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "error using service account to build config: %s")
+	}
+
+	clientScheme := runtime.NewScheme()
+	err = clientgoscheme.AddToScheme(clientScheme)
+	if err = clientgoscheme.AddToScheme(clientScheme); err != nil {
+		return nil, nil, nil, err
+	}
+
+	directClient, err := client.New(cfg, client.Options{Scheme: clientScheme})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return directClient, cfg, clientScheme, nil
+}
+
+// GetAssociatedNode uses the given client to find the name of the node associated with the VM this is running on
+func GetAssociatedNode(c client.Client) (*core.Node, error) {
+	var nodes core.NodeList
+	if err := c.List(context.TODO(), &nodes); err != nil {
+		return nil, err
+	}
+	addrs, err := localInterfaceAddresses()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This section prevents the functions using this from being unit tested, I'd prefer that this section be moved out of the function, and instead `[]net.Addrs`  be passed in.

Reply:
I don't quite understand how the unit tests were running (and passing) without making this change. I made it anyway